### PR TITLE
refactor: simplify state pruning

### PR DESCRIPTION
**Pull request by faddat**
Like #2152, this PR has been checked out of #2130.  

This PR breaks a pruning down into a function with helpers, instead of a single large function.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [ ] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: internal/state/store.go [hunk 1]
```diff
@@ -329,90 +315,11 @@ func (store dbStore) PruneStates(from int64, to int64, evidenceThresholdHeight i
...
-		// avoid batches growing too large by flushing to database regularly
-		if pruned%1000 == 0 && pruned > 0 {
-			err := batch.Write()
-			if err != nil {
-				return err
-			}
-			batch.Close()
-			batch = store.db.NewBatch()
-			defer batch.Close()
-		}
```
Comment (mzabaluev) [inline internal/state/store.go:415]: This goes beyond code simplification... Why was batch size management removed here?

Comment (faddat) [reply internal/state/store.go]: thanks for your review.  That was in error.

**Comment by faddat** [comment]
@mzabaluev fyi, this has been updated to fix the missing batching.  Thanks!

**Comment by faddat** [comment]
this now needs rework, and I'm hesitant to do the merge because of the risk of missing something.

**Pull request closed without merging**